src/backends/torch/torchsolver.h

```diff
@@ -105,6 +106,12 @@ namespace dd
       return _base_lr;
     }
 
+    void swap_swa_sgd()
+    {
+      if (_swa)
+        (reinterpret_cast<Ranger *>(_optimizer.get()))->swap_swa_sgd();
+    }
```

Comment:
With all modifications above we should not require this method anymore.

Reply:
mmmm I do not see how to remove it entirely